### PR TITLE
Gradle: prefer local deps as JARs when building runnable JARs

### LINKED ISSUE #7365: Submodule not in classpath when running :app:quarkusBuild in a multi-module Gradle project
**Describe the bug**

Running **:app:quarkusBuild** in a multi-module project with a shared module and an application module leads to

` java.lang.NoClassDefFoundError: com/demo/common/model/SharedModel
        at com.demo.application.Application.schedule(Application.java:19)
        at com.demo.application.Application_Subclass.schedule$$superaccessor8(Application_Subclass.zig:21)
        at com.demo.application.Application_Subclass$$function$$8.apply(Application_Subclass$$function$$8.zig:29)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:54)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.invokeInOurTx(TransactionalInterceptorBase.java:119)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.invokeInOurTx(TransactionalInterceptorBase.java:92)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired.doIntercept(TransactionalInterceptorRequired.java:32)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.intercept(TransactionalInterceptorBase.java:53)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired.intercept(TransactionalInterceptorRequired.java:26)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired_Bean.intercept(TransactionalInterceptorRequired_Bean.zig:168)
        at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:41)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
        at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:32)
        at com.demo.application.Application_Subclass.schedule(Application_Subclass.zig:86)
        at com.demo.application.Application_ClientProxy.schedule(Application_ClientProxy.zig:51)
        at com.demo.application.Application_ScheduledInvoker_schedule_dce0524c3f1de653f83719900424446459b7c471.invokeBean(Application_ScheduledInvoker_schedule_dce0524c3f1de653f83719900424446459b7c471.zig:45)
        at io.quarkus.arc.runtime.BeanInvoker.invoke(BeanInvoker.java:20)
        at io.quarkus.scheduler.runtime.SimpleScheduler$2.run(SimpleScheduler.java:123)
        at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
        at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1535)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
        at java.base/java.lang.Thread.run(Thread.java:834)
        at org.jboss.threads.JBossThread.run(JBossThread.java:479)
` 

**Expected behavior**

Library classes added to the classpath as well

**Actual behavior**

Classes of the library are **not getting added to the classpath**

**To Reproduce**

Demo Project with the issue and workaround by pre-building the library manually before the application

https://github.com/mklueh/quarkus-multi-module-submodule-classpath-issue

### PR DESCRIPTION
**Pull request by aloubyansky**
When building runnable JARs using Gradle, dependencies from the same project/workspace should be consumed as JARs, not as directories of classes

Fixes #7365

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]
Looks good, but can we add a test too?

**Comment by aloubyansky** [comment]
Added a test.

File: devtools/gradle/src/functionalTest/java/io/quarkus/gradle/BasicMultiModuleTest.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        
+        BuildResult build = GradleRunner.create()
+                .forwardOutput()
+                .withPluginClasspath()
+                .withArguments(arguments(":application:quarkusBuild"))
+                .withProjectDir(projectDir)
+                .build();
+        
+        final Path commonLibs = projectDir.toPath().resolve("common").resolve("build").resolve("libs");
+        assertThat(Files.exists(commonLibs)).isEqualTo(true);
```
Comment (gastaldi) [inline devtools/gradle/src/functionalTest/java/io/quarkus/gradle/BasicMultiModuleTest.java]:
```suggestion
        assertThat(commonLibs).exists();
```

File: devtools/gradle/src/functionalTest/java/io/quarkus/gradle/BasicMultiModuleTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        BuildResult build = GradleRunner.create()
+                .forwardOutput()
+                .withPluginClasspath()
+                .withArguments(arguments(":application:quarkusBuild"))
+                .withProjectDir(projectDir)
+                .build();
+        
+        final Path commonLibs = projectDir.toPath().resolve("common").resolve("build").resolve("libs");
+        assertThat(commonLibs).exists();
+        assertThat(Files.exists(commonLibs.resolve("common.jar"))).isEqualTo(true);
```
Comment (gastaldi) [inline devtools/gradle/src/functionalTest/java/io/quarkus/gradle/BasicMultiModuleTest.java]: Same as above

**Comment by gastaldi** [comment]
I rebased and force-pushed your branch with the test changes. Looks great!

**Pull request merged**